proxy/envoy/ingress.go

```diff
@@ -15,256 +15,222 @@
 package envoy
 
 import (
+	"context"
+	"crypto/sha256"
 	"fmt"
 	"io/ioutil"
+	"net/http"
 	"sort"
 	"strconv"
 	"time"
 
 	"github.com/golang/glog"
-	"github.com/golang/protobuf/proto"
 	"github.com/hashicorp/go-multierror"
 
-	"istio.io/api/proxy/v1/config"
+	proxyconfig "istio.io/api/proxy/v1/config"
 	"istio.io/manager/model"
 	"istio.io/manager/proxy"
 )
 
+const (
+	ingressNode = "ingress"
+	certFile    = "/etc/tls.crt"
+	keyFile     = "/etc/tls.key"
+)
+
 type ingressWatcher struct {
 	agent   proxy.Agent
-	ctl     model.Controller
-	context *IngressConfig
+	secrets model.SecretRegistry
+	mesh    *proxyconfig.ProxyMeshConfig
 }
 
 // NewIngressWatcher creates a new ingress watcher instance with an agent
-func NewIngressWatcher(ctl model.Controller, context *IngressConfig) (Watcher, error) {
-	agent := proxy.NewAgent(runEnvoy(context.Mesh, "ingress"), cleanupEnvoy(context.Mesh), 10, 100*time.Millisecond)
-
+func NewIngressWatcher(mesh *proxyconfig.ProxyMeshConfig, secrets model.SecretRegistry) (Watcher, error) {
+	agent := proxy.NewAgent(runEnvoy(mesh, ingressNode), cleanupEnvoy(mesh), 10, 100*time.Millisecond)
 	out := &ingressWatcher{
 		agent:   agent,
-		ctl:     ctl,
-		context: context,
+		secrets: secrets,
+		mesh:    mesh,
 	}
+	return out, nil
+}
 
-	if err := ctl.AppendConfigHandler(model.IngressRule, func(model.Key, proto.Message, model.Event) {
-		out.reload()
-	}); err != nil {
-		return nil, err
-	}
+func (w *ingressWatcher) Run(stop <-chan struct{}) {
```

Comment:
Yes, I was thinking the same. kubernetes library doesn't use contexts (yet) though...